cg/store/api/base.py

```diff
@@ -1,27 +1,105 @@
 """All models aggregated in a base class"""
-from attr import dataclass
+from typing import Any, Type
 
-from cg.store import models
+from alchy import Query, ModelBase
+from dataclasses import dataclass
+
+from cg.store.models import (
+    Analysis,
+    Application,
+    ApplicationVersion,
+    Bed,
+    BedVersion,
+    Collaboration,
+    Customer,
+    Delivery,
+    Family,
+    FamilySample,
+    Flowcell,
+    Invoice,
+    Organism,
+    Panel,
+    Pool,
+    Sample,
+    User,
+)
 
 
 @dataclass
 class BaseHandler:
-    """All models in one base class"""
-
-    User = models.User
-    Bed = models.Bed
-    BedVersion = models.BedVersion
-    Customer = models.Customer
-    Collaboration = models.Collaboration
-    Sample = models.Sample
-    Family = models.Family
-    FamilySample = models.FamilySample
-    Flowcell = models.Flowcell
-    Analysis = models.Analysis
-    Application = models.Application
-    ApplicationVersion = models.ApplicationVersion
-    Panel = models.Panel
-    Pool = models.Pool
-    Delivery = models.Delivery
-    Invoice = models.Invoice
-    Organism = models.Organism
+    """All models in one base class."""
+
+    Analysis: Type[ModelBase] = Analysis
+    Application: Type[ModelBase] = Application
+    ApplicationVersion: Type[ModelBase] = ApplicationVersion
+    Bed: Type[ModelBase] = Bed
+    BedVersion: Type[ModelBase] = BedVersion
+    Collaboration: Type[ModelBase] = Collaboration
+    Customer: Type[ModelBase] = Customer
+    Delivery: Type[ModelBase] = Delivery
+    Family: Type[ModelBase] = Family
+    FamilySample: Type[ModelBase] = FamilySample
+    Flowcell: Type[ModelBase] = Flowcell
+    Invoice: Type[ModelBase] = Invoice
+    Organism: Type[ModelBase] = Organism
+    Panel: Type[ModelBase] = Panel
+    Pool: Type[ModelBase] = Pool
+    Sample: Type[ModelBase] = Sample
+    User: Type[ModelBase] = User
+
+    @staticmethod
+    def _get_query(table: Type[ModelBase]) -> Query:
+        """Return a query for the given table."""
+        return table.query
+
+    def _get_join_sample_ApplicationVersion(self) -> Query:
```

Comment:
I think this is duplicated with ` def _get_join_sample_application_version_query(self) -> Query:` below

Reply:
Good catch 👀